flows/definition/dependencies.go

```diff
@@ -6,12 +6,13 @@ import (
 )
 
 type dependencies struct {
-	Channels []*assets.ChannelReference `json:"channels,omitempty"`
-	Contacts []*flows.ContactReference  `json:"contacts,omitempty"`
-	Fields   []*assets.FieldReference   `json:"fields,omitempty"`
-	Flows    []*assets.FlowReference    `json:"flows,omitempty"`
-	Groups   []*assets.GroupReference   `json:"groups,omitempty"`
-	Labels   []*assets.LabelReference   `json:"labels,omitempty"`
+	Channels  []*assets.ChannelReference  `json:"channels,omitempty"`
+	Contacts  []*flows.ContactReference   `json:"contacts,omitempty"`
+	Fields    []*assets.FieldReference    `json:"fields,omitempty"`
+	Flows     []*assets.FlowReference     `json:"flows,omitempty"`
+	Groups    []*assets.GroupReference    `json:"groups,omitempty"`
+	Labels    []*assets.LabelReference    `json:"labels,omitempty"`
+	Templates []*assets.TemplateReference `json:"templates,omitempty"`
```

Comment:
It feels like a shame that references aren't self typing in JSON so that we could avoid this. Maybe worth considering at some point?

Reply:
you mean a single merged list and they have a type field, or a dynamic way of building these lists? The latter doesn't sound impossible

Reply:
The former.